src/main/java/org/openrewrite/java/testing/cucumber/CucumberJava8HookDefinitionToCucumberJava.java

```diff
@@ -0,0 +1,218 @@
+            int argumentsSize = arguments.size();
+            Expression lambdaArgument = arguments.get(argumentsSize - 1);
+            HookArguments hookArguments = new HookArguments(
+                    methodName,
+                    null,
+                    null,
+                    (J.Lambda) lambdaArgument);
+            if (argumentsSize == 1) {
+                return hookArguments;
+            }
+
+            J.Literal firstArgument = (J.Literal) arguments.get(0);
+            if (argumentsSize == 2) {
+                // First argument is either a String or an int
+                if (firstArgument.getType() == Primitive.String) {
+                    return hookArguments.withTagExpression((String) firstArgument.getValue());
+                }
+                return hookArguments.withOrder((Integer) firstArgument.getValue());
+            }
+            // First argument is always a String, second argument always an int
+            return hookArguments
+                    .withTagExpression((String) firstArgument.getValue())
+                    .withOrder((Integer) ((J.Literal) arguments.get(1)).getValue());
+        }
+    }
+
+}
+
+@Value
+class HookArguments {
+
+    String annotationName;
+    @Nullable
+    @With
+    String tagExpression;
+    @Nullable
+    @With
+    Integer order;
+    J.Lambda lambda;
+
+    String replacementImport() {
+        return String.format("io.cucumber.java.%s", annotationName);
+    }
+
+    String template() {
+        return "@#{}#{}\npublic void #{}(#{}) throws Exception {\n\t#{any()}\n}";
+    }
+
+    private String formatAnnotationArguments() {
+        if (tagExpression == null && order == null) {
+            return "";
+        }
+        StringBuilder template = new StringBuilder();
+        template.append('(');
+        if (order != null) {
+            template.append("order = ").append(order);
+            if (tagExpression != null) {
+                template.append(", value = \"").append(tagExpression).append('"');
+            }
+        } else {
+            template.append('"').append(tagExpression).append('"');
+        }
+        template.append(')');
+        return template.toString();
+    }
+
+    private String formatMethodName() {
+        return String.format("%s%s%s",
+                annotationName
+                        .replaceFirst("^Before", "before")
+                        .replaceFirst("^After", "after"),
+                tagExpression == null ? ""
+                        : "_tag_" + tagExpression
+                                .replaceAll("[^A-Za-z0-9]", "_"),
+                order == null ? "" : "_order_" + order);
+    }
+
+    private String formatMethodArguments() {
+        J firstLambdaParameter = lambda.getParameters().getParameters().get(0);
+        if (firstLambdaParameter instanceof J.VariableDeclarations) {
+            return String.format("io.cucumber.java.Scenario %s",
```

Comment:
Fully qualified class name to work around not yet knowing how else to insert the correct type.
Could be either properly fixed here, or through a `doAfterVisit` to go from FQCN use to import.